src/modules/task/task.service.ts

```diff
@@ -1,23 +1,124 @@
+import { redisTaskKey, redisTasksKey } from '../../cache/redis.keys';
+import { RedisService } from '../../cache/redis.service';
+import { TaskEntity } from '../../database/entities/task.entity';
+import { UserEntity } from '../../database/entities/user.entity';
+import { ConflictException, NotFoundException } from '../../exceptions';
 import logger from '../../logger';
-import { CreateTaskDto, PaginationDto } from './dto';
+import { PaginationDto } from '../../shared';
+import { CreateTaskDto } from './dto';
 
 @injectable()
 export class TaskService {
-  getTasks(dto: PaginationDto) {
-    logger.info(`Чтение списка задач`);
+  constructor(
+    @inject(RedisService)
+    private readonly redis: RedisService,
+  ) {}
 
-    return { limit: `${dto.limit}` };
+  async getTasks(dto: PaginationDto) {
+    logger.info(`Запрос на чтение списка задач`);
+
+    const { limit, offset } = dto;
+
+    const cacheTasks = await this.redis.get<TaskEntity>(redisTasksKey(limit, offset));
+
+    if (cacheTasks) {
+      return cacheTasks;
+    }
+
+    const tasks = await TaskEntity.findAll({
+      limit,
+      offset,
+      attributes: ['title', 'status', 'importance', 'description'],
+      include: [{ model: UserEntity, attributes: ['id', 'name'] }],
+    });
+    if (!tasks) {
+      throw new NotFoundException('Задач не найдено');
+    }
+
+    await this.redis.set(redisTasksKey(limit, offset), tasks, { EX: 300 });
+
+    return tasks;
   }
 
-  getTaskById(id: number) {
-    logger.info(`Чтение задачи по id=${id}`);
+  async getTaskById(idTask: TaskEntity['id']) {
+    logger.info(`Чтение задачи по id=${idTask}`);
+
+    const cacheTask = await this.redis.get<TaskEntity>(redisTaskKey(idTask));
+
+    if (cacheTask) {
+      return cacheTask;
+    }
 
-    return { message: `Вы пытаетесь прочитать задачу id=${id}` };
+    const findById = await TaskEntity.findOne({
+      where: { id: idTask },
+      attributes: ['title', 'status', 'importance', 'description'],
+      include: [{ model: UserEntity, attributes: ['id', 'name'] }],
+    });
+
+    if (!findById) {
+      throw new NotFoundException('Такой задачи не найдено');
+    }
+
+    await this.redis.set(redisTaskKey(idTask), findById, { EX: 300 });
+
+    return { findById };
   }
-  createTask(dto: CreateTaskDto) {
+
+  async createTask(dto: CreateTaskDto) {
     logger.info(`Создание задачи`);
 
+    const task = await TaskEntity.findOne({
+      where: { title: dto.title },
+    });
+
+    if (task) {
+      throw new ConflictException('Такая задача уже есть');
+    }
+
+    await this.findUser(dto.assigneeId);
+
+    const newTask = await TaskEntity.create({
+      title: dto.title,
+      status: dto.status,
+      description: dto.description,
+      importance: dto.importance,
+      assigneeId: dto.assigneeId,
+    });
+
     return { message: `Вы создали задачу ${dto.title}` };
   }
+
+  async updateTask(dto: CreateTaskDto, idTask: TaskEntity['id']) {
```

Comment:
```suggestion
  async updateTask(dto: UpdateTaskDto, idTask: TaskEntity['id']) {
```

Создать UpdateTaskDto - это как create, но абсолютно все поля опциональные